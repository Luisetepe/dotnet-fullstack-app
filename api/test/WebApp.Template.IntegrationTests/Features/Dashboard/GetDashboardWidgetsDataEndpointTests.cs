using WebApp.Template.Api.Endpoints.Dashboard;

namespace WebApp.Template.IntegrationTests.Features.Dashboard
{
    using System.Threading.Tasks;
    using Microsoft.EntityFrameworkCore;
    using Snapshooter.Xunit;
    using WebApp.Template.Application.Features.Dashboard.Queries.GetDashboardWidgetsData;
    using Xunit;

    public class GetDashboardWidgetsDataEndpointTests(IntegrationTestFixture fixture)
        : TestBase<IntegrationTestFixture>
    {
        [Fact]
        public async Task Should_Return_Dashboard_Widgets_Data()
        {
            // Act
            var (httpResponse, result) = await fixture.Client.GETAsync<
                GetDashboardWidgetsDataEndpoint,
                GetDashboardWidgetsDataResponse
            >();

            // Assert
            httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);

            result.Should().NotBeNull();
            result.Should().MatchSnapshot(matchOptions => matchOptions.IgnoreFields("**.Id"));
        }
    }
}
