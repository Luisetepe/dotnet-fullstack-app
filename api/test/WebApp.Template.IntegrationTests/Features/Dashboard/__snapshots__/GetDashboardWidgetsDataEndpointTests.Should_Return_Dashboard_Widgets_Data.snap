{
  "Plants": 5,
  "Locations": 5,
  "SolarCapacity": 1200.0,
  "StorageCapacity": 8000.0
}
